internal/bminventory/inventory.go

```diff
@@ -3619,6 +3616,12 @@ func (b *bareMetalInventory) listClustersInternal(ctx context.Context, params in
 			// Clear this field as it is not needed to be sent via API
 			h.FreeAddresses = ""
 		}
+		schedulableMasters, err := b.providerRegistry.GetActualSchedulableMasters(c)
+		if err != nil {
+			log.WithError(err).Error("failed to retrieve the actual schedulable masters value")
+			return nil, common.NewApiError(http.StatusInternalServerError, err)
+		}
+		c.SchedulableMasters = &schedulableMasters
```

Comment:
We don't typically make such on-request adjustments to data returned from the database, could this property be stored in the database somehow rather than computed on each request?

Reply:
By default the "schedulable_masters" flag is not set and we need to respond with the expected value depending on the provider/number of active hosts, etc. Having the actual value saved in the database would need the database to be updated with any state change (each time the number of hosts or the provider changed...). I don't think doing the computation and saving it on every state change is better than just compute the actual value on the fly when needed.

Reply:
I remember @avishayt bringing up this concern about a year ago on one of my PRs, I'm not entirely sure what's the motivation behind this constraint, maybe he can weigh in

Reply:
I think I agree with @mkowalski  and @omertuc  here. I would rather have the data properly reflected in the DB rather than dynamically computed before handing it off to `openshift-insdtall` The motivation is that we are not the ones doing the deployment itself and there are some decisions being made dynamically on the `openshift-installer` side already.

Reply:
> Thanks for the PR and the patience. I'm sorry it took this long to get back to it.
> 
> Could you please share an example of what the final `install-config` looks like when masters are schedulable?

This implementation does not touch/alter the `install-config.yaml`, it just adds an additional manifest (`50-schedulable_masters.yaml`) to the installation to enable/disable the scheduling.
Something like:
```
apiVersion: config.openshift.io/v1
kind: Scheduler
metadata:
  name: cluster
spec:
  mastersSchedulable: true
  policy:
    name: ""
status: {}
```

This is because the current implementation does not allow to effectively disable the scheduling if needed. It just adds and additional manifest to enable the scheduling if `schedulable_masters=true`, otherwise it does nothing and the default value generated by the 'openshift-install' command is used (which is different depending on the number of hosts in the cluster). 

The idea is to use the expected defaults for each platform if `schedulable_masters=nil` (the default) or the value saved in the database otherwise (the user explicitly wants to enable/disable the scheduling).

On the other hand, we need to disable the scheduling in **oVirt** and **vSphere** providers by default, no matter the number of hosts in the cluster. This is because we set the number of compute replicas to 0 in the install config to prevent the MCO trying to provision new worker nodes and has the side effect of enabling the scheduling on masters.

Reply:
> I think I agree with @mkowalski and @omertuc here. I would rather have the data properly reflected in the DB rather than dynamically computed before handing it off to `openshift-insdtall` The motivation is that we are not the ones doing the deployment itself and there are some decisions being made dynamically on the `openshift-installer` side already.

One of the motivation of this change is that we need to return the real value of the `schedulable_masters` property on each call (for example the UI needs to know it to show the correct information, now it's using a workaround to fix this). That real value depends on the number of hosts and it varies over time, so lets say `schedulable_masters=nil` (the user didn't actively set the property at cluster creation time) and we have three masters. In this case the expected value is `true`. Then the user adds one more host so now the expected value is `false`. If the user configured the property, then the saved value in the database must be returned no matter how many hosts are in the cluster. 

I guess this could be done by computing the expected default value when a new host is added/removed from the cluster and updating the database accordingly or do nothing if the user explicitly set the property. Would you prefer this approach? I am not really sure about the implications of this and I would probably need some help to get this implemented in this way.

Reply:
>so lets say schedulable_masters=nil (the user didn't actively set the property at cluster creation time) and we have three masters. In this case the expected value is true. Then the user adds one more host so now the expected value is false. If the user configured the property, then the saved value in the database must be returned no matter how many hosts are in the cluster.

We already do similar updates in the network options. This is a very natural behaviour - as the state of the cluster changes (by e.g. adding or removing hosts), some of the properties need to be recalculated. That's perfectly fine and it's not a bad pattern at all.

One more important thing here. We should have proper and reasonable defaults. If "user didn't actively set the property at cluster creation time" we should not just put `nil` to the DB, but we should agree on a default value and roll like this. You can use `disk_encryption` property as an example - we have introduced defaults there and have explicit code that handles the scenario when user did not provide value at all. But in principle, if we are introducing on/off field, we should
1) create a migration that will backfill the DB for already existing clusters
2) only allow true/false values, because `nil` has no semantics for a feature that can be either enabled or disabled